### PR TITLE
refactor: cleanup request-node

### PR DESCRIPTION
**Pull request by benjlevesque**

### CODE REVIEW

File: packages/request-node/src/requestNode.ts [hunk 1]
```diff
@@ -162,93 +174,46 @@ class RequestNode {
...
-          clientRequest,
-          serverResponse,
-          this.dataAccess,
-          this.logger,
-        );
-      } else {
-        return serverResponse.status(httpStatus.SERVICE_UNAVAILABLE).send(NOT_INITIALIZED_MESSAGE);
-      }
-    });
+    router.get('/healthz', (_, res) => res.status(StatusCodes.OK).send('OK'));
```
Comment (vrolland) [inline packages/request-node/src/requestNode.ts:177]: :heart_eyes:

File: packages/request-node/src/requestNode.ts [hunk 2]
```diff
@@ -162,93 +174,46 @@ class RequestNode {
...
+    router.use(this.initializedMiddelware());
+    router.get('/readyz', (_, res) => res.status(StatusCodes.OK).send('OK'));
+    router.get('/status', this.getStatusHandler.handler);
+    router.post('/ipfsAdd', this.ipfsAddHandler.handler);
+    router.post('/persistTransaction', this.persistTransaction.handler);
+    router.get('/getConfirmedTransaction', this.confirmedTransactionStore.getConfirmedTransaction);
```
Comment (vrolland) [inline packages/request-node/src/requestNode.ts:183]: Just curiousity, why there is no `handler` here?

**Review by vrolland — APPROVED** [review]
Was hard to CR at the beginning. But looks super neat :) Well done!

File: packages/request-node/src/requestNode.ts [hunk 3]
```diff
@@ -162,93 +174,46 @@ class RequestNode {
...
+  }
 
-    // Route for getChannelsByTopic request
-    router.get('/getChannelsByTopic', (clientRequest: any, serverResponse: any) => {
-      if (this.initialized) {
-        return getChannelsByTopic(clientRequest, serverResponse, this.dataAccess, this.logger);
+  /**
+   * Middleware to refuse traffic if node is not initialized yet
+   */
+  private initializedMiddelware() {
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:209]: 💯

File: packages/request-node/src/requestNode.ts [hunk 4]
```diff
@@ -2,20 +2,22 @@ import { DataAccess, TransactionIndex } from '@requestnetwork/data-access';
...
 import { getCustomHeaders, getInitializationStorageFilePath, getMnemonic } from './config';
 import ConfirmedTransactionStore from './request/confirmedTransactionStore';
-import getChannelsByTopic from './request/getChannelsByTopic';
+import GetChannelsByTopicHandler from './request/getChannelsByTopic';
 import getStatus from './request/getStatus';
-import getTransactionsByChannelId from './request/getTransactionsByChannelId';
-import ipfsAdd from './request/ipfsAdd';
+import GetTransactionsByChannelIdHandler from './request/getTransactionsByChannelId';
+import IpfsAddHandler from './request/ipfsAdd';
 import PersistTransaction from './request/persistTransaction';
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:16]:
```suggestion
import PersistTransactionHandler from './request/persistTransaction';
```

File: packages/request-node/src/requestNode.ts [hunk 5]
```diff
@@ -76,8 +83,19 @@ class RequestNode {
...
-    this.persistTransaction = new PersistTransaction(this.confirmedTransactionStore);
+    this.confirmedTransactionStore = new ConfirmedTransactionStore(this.logger, store);
+    this.getTransactionsByChannelIdHandler = new GetTransactionsByChannelIdHandler(
+      this.logger,
+      this.dataAccess,
+    );
+    this.getChannelByTopicHandler = new GetChannelHandler(this.logger, this.dataAccess);
+    this.getStatusHandler = new GetStatusHandler(this.logger, this.dataAccess);
+    this.ipfsAddHandler = new IpfsAddHandler(this.logger, this.ethereumStorage);
+    this.persistTransaction = new PersistTransaction(
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:94]:
```suggestion
    this.persistTransactionHandler = new PersistTransactionHandler(
```

File: packages/request-node/src/requestNode.ts [hunk 6]
```diff
@@ -39,12 +41,17 @@ class RequestNode {
   public dataAccess: DataAccess;
   public ethereumStorage: StorageTypes.IStorage;
 
-  private express: express.Express;
+  private express: express.Application;
   private initialized: boolean;
   private logger: LogTypes.ILogger;
   private persistTransaction: PersistTransaction;
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:47]:
```suggestion
  private persistTransactionHandler: PersistTransactionHandler;
```

File: packages/request-node/src/requestNode.ts [hunk 7]
```diff
@@ -162,93 +174,46 @@ class RequestNode {
...
-      } else {
-        return serverResponse.status(httpStatus.SERVICE_UNAVAILABLE).send(NOT_INITIALIZED_MESSAGE);
-      }
-    });
+    router.get('/healthz', (_, res) => res.status(StatusCodes.OK).send('OK'));
+    router.use(this.initializedMiddelware());
+    router.get('/readyz', (_, res) => res.status(StatusCodes.OK).send('OK'));
+    router.get('/status', this.getStatusHandler.handler);
+    router.post('/ipfsAdd', this.ipfsAddHandler.handler);
+    router.post('/persistTransaction', this.persistTransaction.handler);
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:182]:
```suggestion
    router.post('/persistTransaction', this.persistTransactionHandler.handler);
```

File: packages/request-node/src/requestNode.ts [hunk 8]
```diff
@@ -2,20 +2,22 @@ import { DataAccess, TransactionIndex } from '@requestnetwork/data-access';
...
+import express, { NextFunction, Request, Response } from 'express';
+import { StatusCodes } from 'http-status-codes';
 import KeyvFile from 'keyv-file';
 
 import Utils from '@requestnetwork/utils';
 import { getCustomHeaders, getInitializationStorageFilePath, getMnemonic } from './config';
 import ConfirmedTransactionStore from './request/confirmedTransactionStore';
-import getChannelsByTopic from './request/getChannelsByTopic';
+import GetChannelsByTopicHandler from './request/getChannelsByTopic';
 import getStatus from './request/getStatus';
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:13]: Already declared bellow
```suggestion
```

File: packages/request-node/src/requestNode.ts [hunk 9]
```diff
@@ -2,20 +2,22 @@ import { DataAccess, TransactionIndex } from '@requestnetwork/data-access';
...
-import getTransactionsByChannelId from './request/getTransactionsByChannelId';
-import ipfsAdd from './request/ipfsAdd';
+import GetTransactionsByChannelIdHandler from './request/getTransactionsByChannelId';
+import IpfsAddHandler from './request/ipfsAdd';
 import PersistTransaction from './request/persistTransaction';
 import { getEthereumStorage } from './storageUtils';
 import { Server } from 'http';
+import GetChannelHandler from './request/getChannelsByTopic';
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:19]: Already declared above
```suggestion
```

File: packages/request-node/src/requestNode.ts [hunk 10]
```diff
@@ -39,12 +41,17 @@ class RequestNode {
...
+  private express: express.Application;
   private initialized: boolean;
   private logger: LogTypes.ILogger;
   private persistTransaction: PersistTransaction;
   private confirmedTransactionStore: ConfirmedTransactionStore;
   private requestNodeVersion: string;
+
+  private getTransactionsByChannelIdHandler: GetTransactionsByChannelIdHandler;
+  private getChannelByTopicHandler: GetChannelsByTopicHandler;
+  private getStatusHandler: getStatus;
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:53]:
```suggestion
  private getStatusHandler: getStatusHandler;
```

File: packages/request-node/src/requestNode.ts [hunk 11]
```diff
@@ -76,8 +83,19 @@ class RequestNode {
...
     });
 
-    this.confirmedTransactionStore = new ConfirmedTransactionStore(store);
-    this.persistTransaction = new PersistTransaction(this.confirmedTransactionStore);
+    this.confirmedTransactionStore = new ConfirmedTransactionStore(this.logger, store);
+    this.getTransactionsByChannelIdHandler = new GetTransactionsByChannelIdHandler(
+      this.logger,
+      this.dataAccess,
+    );
+    this.getChannelByTopicHandler = new GetChannelHandler(this.logger, this.dataAccess);
```
Comment (alexandre-abrioux) [inline packages/request-node/src/requestNode.ts:91]:
```suggestion
    this.getChannelByTopicHandler = new GetChannelByTopicHandler(this.logger, this.dataAccess);
```

**Review by alexandre-abrioux — APPROVED** [review]
LGTM!

Made two suggestions:
1. rename `PersistTransaction` to `PersistTransactionHandler` in the import of `requestNode.ts` to align with the class name
2. remove double imports in the `requestNode.ts` (`./request/getStatus` and `./request/getChannelsByTopic` are imported twice)

File: packages/request-node/src/requestNode.ts (hunk 2 above)
Comment (benjlevesque) [reply packages/request-node/src/requestNode.ts:183]: I hesitated, actually all other handlers are based on this one 
but this class is a "store" for a given type of data

I was thinking to later introduce a Handler for this endpoint, separated from the Store

Comment (benjlevesque) [reply packages/request-node/src/requestNode.ts]: actually I did this change ;)

**Pull request merged**